scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/vfs/Paths.scala

```diff
@@ -32,11 +30,6 @@ object Paths {
 
   def scalaToSemanticdb(path: RelativePath): RelativePath = {
     require(isScala(path))
-    RelativePath(
-      semanticDbPrefix +
-        path.toString
-          .stripPrefix(scalaPrefix)
-          .stripSuffix(scalaSuffix) +
-        semanticDbSuffix)
+    semanticDbPrefix.resolve(path.resolveSibling(_.stripSuffix(scalaSuffix) + semanticDbSuffix))
```

Comment:
I've refactored the logic a bit to make it cleaner. I agree my refactoring was a bit messy. I did not like the previous implementation since it was purely string based. A path is more a `List[String]` than a `String`, so when changing the extension we should try to use `resolveSibling` and when checking the extension we should check the extension against the filename instead of `Path.toString.endsWith`.